### PR TITLE
Report `RhpInitialDynamicInterfaceDispatch` reference in gfids

### LINKED ISSUE #103798: Failure in the Exceptions smoke test with CFG
When running the Exceptions test with CFG, control flow guard kills the process when we try to call `RhpInitialInterfaceDispatch`. We need to investigate what's going on. This method is address taken in the native runtime so it should be added to the bitmask by the C++ compiler. Also it's suspicious interface dispatch would be broken when we run ILC in the CI with CFG enabled.

### PR DESCRIPTION
**Pull request by MichalStrehovsky**
Some subset of tests is failing with control flow guard enabled because we're not able to indirectly call `RhpInitialDynamicInterfaceDispatch`. It's not clear why since we address take this method in C++ source files as well, but we should be reporting this from our object file too and doing that fixes it.

Fixes #103798

Cc @dotnet/ilc-contrib

### CODE REVIEW

File: src/coreclr/jit/hwintrinsicxarch.cpp [hunk 1]
```diff
@@ -267,17 +267,17 @@ static CORINFO_InstructionSet lookupInstructionSet(const char* className)
         {
             if (strncmp(className + 6, "128", 3) == 0)
             {
-                assert((className[9] == '\0') || (strcmp(className + 9, "`1") == 0));
+                //assert((className[9] == '\0') || (strcmp(className + 9, "`1") == 0));
```
Comment (filipnavara) [inline src/coreclr/jit/hwintrinsicxarch.cpp:270]: Unintended change?

Comment (MichalStrehovsky) [reply src/coreclr/jit/hwintrinsicxarch.cpp:270]: See commit title https://github.com/dotnet/runtime/pull/103948/commits/f71b07387ae4fc43b01192253945b79e46f58275

Outerloop is on the floor again and I'm getting really tired being the janitor.

Comment (filipnavara) [reply src/coreclr/jit/hwintrinsicxarch.cpp:270]: GitHub UI only showed me the first two commits in the list and thus I didn't see the title. Must have been some glitch.

File: src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/NodeFactory.cs [hunk 2]
```diff
@@ -44,6 +44,11 @@ public abstract partial class NodeFactory
             ObjectDataInterner dataInterner)
         {
             _target = context.Target;
+
+            InitialInterfaceDispatchStub = _target.Architecture == TargetArchitecture.ARM
+                ? new InitialInterfaceDispatchStubNode()
```
Comment (jkotas) [inline src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/NodeFactory.cs:49]: I am wondering whether the special ARM dispatch stub wrapper is still necessary.

**Review by jkotas — APPROVED** [review]
LGTM (with workarounds reverted)

File: src/coreclr/jit/hwintrinsicxarch.cpp (hunk 1 above)
Comment (tannergooding) [reply src/coreclr/jit/hwintrinsicxarch.cpp:270]: A fix for this up here: https://github.com/dotnet/runtime/pull/103995/files

However, the fact that the assert triggered looks to be a bug in NAOT, as there are types being marked as `[Intrinsic]` which should not be marked `[Intrinsic]` and which are not marked as such in the actual managed implementation.

File: src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/NodeFactory.cs (hunk 2 above)
Comment (MichalStrehovsky) [reply src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/NodeFactory.cs:49]: I can try to delete that in a different PR and see what the CI says, but I don't have hardware to test it on.

**Comment by MichalStrehovsky** [comment]
/ba-g only deleted old workarounds and #103215 needs this and I'm going to bed

File: src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/NodeFactory.cs (hunk 2 above)
Comment (jkotas) [reply src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/NodeFactory.cs:49]: Testing it in CI should be good enough if you would like to give it a try.

I think this was only needed for multi-.dll Windows arm32 support that we do not have anymore.

**Pull request merged**